### PR TITLE
Adhoc adding sso and sap credentials

### PR DESCRIPTION
**Pull request by OmarTuffaha**
so it can be put from the main.yml and inserted in projects

### CODE REVIEW

File: templates/etc/env.php.j2 [hunk 1]
```diff
@@ -130,6 +130,32 @@ return array (
...
+  {% if magento_queue_amqp is defined%}
+    {% if magento_sso_oauth_credentials %}
+    'websso_oauth' =>
+    array (
+      'domain' => '{{ magento_sso_oauth_credentials.domain }}',
+      'client_id' => '{{ magento_sso_oauth_credentials.client_id }}',
+      'client_secret' => '{{ magento_sso_oauth_credentials.client_secret }}',
+    ),
+    {% endif %}
+    {% if magento_sap_oauth_credentials %}
```
Comment (aario-sitewards) [inline templates/etc/env.php.j2]: please change it to magento_sap_credentials

**Comment by aboritskiy** [comment]
this role is a generic magento2 role, while solution provided is only suitable for one specific project we are working on. Can we generalize the solution?

File: templates/etc/env.php.j2 [hunk 2]
```diff
@@ -130,6 +130,32 @@ return array (
   {% if magento_run_message_queue_consumers_separately %}
   'cron_consumers_runner' => array(
       'cron_run' => false
-  )
+  ),
+  {% endif %}
+  {% if magento_queue_amqp is defined%}
```
Comment (aboritskiy) [inline templates/etc/env.php.j2]: here seem to be a mistake, you duplicated the `magento_queue_amqp`  condition, which has very little to do with SSO/SAP credentials

**Comment by aboritskiy** [comment]
suggestion: 
1. Add a new variable called: `magento_additional_configuration` or alike
2. Make that variable empty by default
3. Just print contents of that variable directly in the `env.php` at the very end.
4. In the actual project (Zxxxxx) fill the variable with appropriate content needed for that project using multiline syntax:
```
magento_additional_configuration: |
    here is your multiline content,
    and more
    and more 
    and more
```

**Pull request merged**